### PR TITLE
MPDP 542 Download CSV Link

### PR DESCRIPTION
**Pull request by Afshin-Karamifar-Capgemini**

### CODE REVIEW

File: app/routes/downloadPaymentsByYearSummary.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  path: '/downloadPaymentsByYearSummary',
+  handler: async (_request: Request, _response: ResponseToolkit) => {
+    try {
+      const content = await utils.getBuffer(urlcsv);
+      return _response
+        .response(content)
+        .type('application/csv')
+        .header(
+          'Content-Disposition',
+          'attachment; filename="ffc-year-payments-summary.csv"'
```
Comment (SidG-CG) [inline app/routes/downloadPaymentsByYearSummary.ts:18]: can you change the filename to ffc-payments-by-year.csv please? This is so it sounds similar to the this api route as well as the FE page.

Comment (Afshin-Karamifar-Capgemini) [reply app/routes/downloadPaymentsByYearSummary.ts]: Thanks, I fixed it.

File: app/routes/downloadPaymentsByYearSummary.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi';
+import config from '../config';
+import * as utils from '../utils';
+
+const urlcsv = `${config.backendEndpoint}` + '/downloadPaymentsByYearSummary';
```
[the remark this answers is not in the document]
Comment (Afshin-Karamifar-Capgemini) [reply app/routes/downloadPaymentsByYearSummary.ts]: Thanks, I fixed it.

File: test/integration/narrow/routes/downloadPaymentsByYearSummary.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const res = await global.__SERVER__.inject(request);
+    expect(res.statusCode).toBe(200);
+  });
+
+  test('GET /downloadPaymentsByYearSummary returns attachment', async () => {
+    const res = await global.__SERVER__.inject(request);
+    expect(res.headers).toHaveProperty('content-type', 'application/csv');
+    expect(res.headers).toHaveProperty(
+      'content-disposition',
+      'attachment; filename="ffc-year-payments-summary.csv"'
```
[the remark this answers is not in the document]
Comment (Afshin-Karamifar-Capgemini) [reply test/integration/narrow/routes/downloadPaymentsByYearSummary.test.ts]: Thanks, I fixed it.

**Review by SidG-CG — APPROVED** [review]

**Pull request merged**